lib/rpmts.c

```diff
@@ -621,7 +621,14 @@ rpmRC rpmtsImportPubkey(const rpmts ts, const unsigned char * pkt, size_t pktlen
         if (krc != RPMRC_OK) {
             rpmlog(RPMLOG_ERR, "%s\n", lints);
         } else {
-            rpmlog(RPMLOG_WARNING, "%s\n", lints);
+            // Don't show the warning if --quiet was provided.  With
+            // --quiet, RPMLOG_WARNING messages are still shown, but
+            // RPMLOG_NOTICE are not.  We can't use RPMLOG_NOTICE
+            // here, as the output needs to go to stdderr.  So, we
+            // check the warning level explicitly.
+            if (rpmIsNormal()) {
+                rpmlog(RPMLOG_WARNING, "%s\n", lints);
```

Comment:
After way too much time and pondering (sorry about that):

This would set a bad precedent, nak. Either it's a warning and needs to be output on all normal verbosity levels, or it's something else, in which case it must not use RPMLOG_WARNING. That the other non-error levels output to stdout is a general rpm limitation (see #962) that we'll just have to live with.

One possible way around this would be adding a separate --nolint or such option which would make pgpPubkeyLint() not called at all, which would then suppress the warning. Only I'm not sure where to stick such a flag, rpmVSFlags_e maybe :thinking:

Reply:
I'm also not so happy with my proposed solution.  `--nolint` seems like a small improvement, but is still not ideal.  I think this is only needed for the test suite, so it is probably best to just deal with it in the test suite and not expose new API.  Or do you envision other places where it is necessary or useful to suppress the lints?

Reply:
Let me put it this way: I can't recall ever regretting adding a disabler for something (in rpm) even if there's no immediately obvious need, but not adding one, several times :sweat_smile: 
As the lint can actually cause the operation to fail, it'd actually be good to have a disabler for it, regardless of the test-suite issue. Biggest problem is finding out a good place for it, neither vsflags or main transaction flags seem quite appropriate.

In the meanwhile, just dealing with the output difference seems like a reasonable way forward.

Reply:
> In the meanwhile, just dealing with the output difference in the test-suite seems like a reasonable way forward.

So this patch should just be dropped, right?

Reply:
Either we need a `--no-lint` option or the test needs to deal with the difference in output.  (I'm currently on vacation and will be back the beginning of August.)

Reply:
I added the `--nolint` option to `rpmkeys` by using a flag in `rpmVSFlags_e`, as @pmatilai suggested.